### PR TITLE
intial port to meson still have lots to do

### PR DESCRIPTION
**Pull request by SolarAquarion**
We do need .sln's, and vcxproj's because meson creates those

---
This change is [Reviewable](https://reviewable.io/reviews/highperformancecoder/classdesc/1)

### CODE REVIEW

**Comment by highperformancecoder** [comment]
I have created a "meson" branch, and merged this pull request into that. Please use that for future pull requests.

File: meson.build [hunk 1]
```diff
@@ -4,7 +4,7 @@ project('classdesc', 'c',
     meson_version: '>=0.39.1',
     license: [ 'GPL3+', 'MIT' ],
     default_options: [
-        'c_std=gnu11',
+        'c_std=gnu99',
```
Comment (highperformancecoder) [inline meson.build]: This is a C++ project, and the version to target is c++98.

There is one test directory that tests classdesc compatibility with more recent C++ standards, currently set to C++11, but due to be updated to C++17 shortly.

**Comment by SolarAquarion** [comment]
cpp_std=c++98

Then?

File: meson.build [hunk 2]
```diff
@@ -0,0 +1,15 @@
+
+project('classdesc', 'c',
+    version: '3.34',
```
Comment (highperformancecoder) [inline meson.build]: Needs to be auto generated from the makeRelease.sh script.

**Comment by SolarAquarion** [comment]
It can probably be a sed job to update that line

**Comment by highperformancecoder** [comment]
Needs to pass Travis CI builds (see .travis.yml), including the full regression suite before being merged into master.

**Comment by highperformancecoder** [comment]
Closing PR now, as the code is merged.

**Comment by SolarAquarion** [comment]
cpp_std=c++11


On Nov 26, 2017 6:14 PM, "Solomon Choina" <shlomochoina@gmail.com> wrote:

It can probably be a sed job to update that line

**Comment by highperformancecoder** [comment]
Yes - sed is a suitable tool.

**Comment by SolarAquarion** [comment]
I have no idea how to do it really

**Comment by highperformancecoder** [comment]
On Sun, Nov 26, 2017 at 11:24:10PM +0000, SolarAquarion wrote:
> I have no idea how to do it really

No worries, it's quite simple - stick it on the todo list (create a
github issue perhaps), and I'll get to it later.

I'm more interested in seeing how meson can autoupdate its dependency
list.

Cheers
-- 

----------------------------------------------------------------------------
Dr Russell Standish                    Phone 0425 253119 (mobile)
Principal, High Performance Coders
Visiting Senior Research Fellow        hpcoder@hpcoders.com.au
Economics, Kingston University         http://www.hpcoders.com.au
----------------------------------------------------------------------------

**Pull request closed without merging**